src/interfaces.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any, Protocol
+
+
+class IComponent(Protocol):
+    logger: "ILogger"
```

Comment:
When exactly do you use "" in TypeHints?

Reply:
I can think of two situations where string type hints are used.

1. When the type doesn't exist yet where the hint is used. That is the case for example when a class has a reference to itself:

```python
class Foo:
    other_foo: "Foo"
```

Because `Foo` doesn't exist yet during its own definition you have to use a string for type hinting.

2. When the type isn't defined yet.

Here the reason is the second case. `ILogger` doesn't exist yet. Therefore I use a string. When `ILogger` is defined I will replace the string with the real type.